pkg/network/protocols/http2/model_linux.go

```diff
@@ -146,27 +146,55 @@ func (tx *EbpfTx) Method() http.Method {
 	}
 }
 
-// StatusCode returns the HTTP status code of the transaction.
+// StatusCode returns the status code of the transaction.
+// If the status code is indexed, then we return the corresponding value.
+// Otherwise, f the status code is huffman encoded, then we decode it and convert it from string to int.
+// Otherwise, we convert the status code from byte array to int.
 func (tx *EbpfTx) StatusCode() uint16 {
-	switch tx.Stream.Response_status_code {
-	case uint16(K200Value):
-		return 200
-	case uint16(K204Value):
-		return 204
-	case uint16(K206Value):
-		return 206
-	case uint16(K400Value):
-		return 400
-	case uint16(K500Value):
-		return 500
-	default:
+	if tx.Stream.Status_code.Indexed_value != 0 {
+		switch tx.Stream.Status_code.Indexed_value {
+		case K200Value:
+			return 200
+		case K204Value:
+			return 204
+		case K206Value:
+			return 206
+		case K400Value:
+			return 400
+		case K500Value:
+			return 500
```

Comment:
what about the range of differnet status code between those values? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
you're looking at the code the handles the status codes that appear in the static table.
Any other status code will be passed as literal for the first time, and since then, will be in the dynamic table

Reply:
Oh I see